### DISCUSSION TITLE
Outpainting suggestion

### DISCUSSION THREAD

**Discussion by guyko81** [hlky/stable-diffusion-webui#979] [Feature suggestions]
Considering that the inpainting functionality is already available, I think it would be really easy to implement the real outpainting: simply generating 4 images, each starts from part of the original image, plus a mask on its edge.
It could go in a rotating clock manner:
![image](https://user-images.githubusercontent.com/10399767/189499488-89a22d0d-c30d-4188-a8b3-96500aa90a39.png)

1. 1st image would be the bottom-right side of the original image (384x384 pixels, which is (512-128)x(512-128) ) plus a mask that takes the shape that fills the 512x512 size, so an L-shape mask of 2 overlapping rectangles of 512x128 size. That would be the bottom-right corner of the image.
2. 2nd image would start from the newly created bottom-right part, but the new mask would only be a top mask, with a L shape of 512x128 + 256x128
3. 3rd image would be generated with a mask with L-shape 256x128 + 512x128
4. 4th image would be with an L-shape mask 256x128 + 256x128

What do you think?

**Comment by Tollanador** [comment]
I think there is a design choice of just _how_ to initialise the tiled images. Just use Perlin, or replicate N number of rows/columns of the original image extending into the outpainted area, or something else.
Then what to do about seams of the original image, plus the 4 created ones, do you then resynthesise the entire new image with a Strength of  0.2 or something? Enough to clean up the seams and keep the majority of the original image.

Do you use the same seed and same prompt for the out-painted sections, or use a random seed and separate prompts?

It should be pretty straight-forward to make the basic 4 tile outpainting using perlin initiation and masking the known original section.

Then just go from there?

**Comment by guyko81** [comment]
Well, I think the Perlin should work with large sampling number, but it should be tested. I think this choice is easily testable. 
Then while generating the individual images I think some level of blurring would be helpful on the meeting edges, to make sure that the colours match with the original image.
The seed shouldn't matter, as we're generating new images, so each seed should lead to a valid image.
The prompt for the outpainted image - if we're talking about a small outpainting, like in my suggestion - should be the same as the original prompt, as most of the part of the newly generated images are coming from the original image. 
However if the outpainting can go beyond this level, then a free prompt would be beneficial, as the newly generated images most probably won't cover the original content, like if I want to create a room around a person then after a certain point my original prompt would be misleading. 
So the prompt should be customizable. 

To be honest a full outpainting experience would be the best, where each new box could be generated automatically by selecting a box that partly covers the original image, and a new prompt could be added. But a simple "upscaling" could be done with the Perlin+rotating+original prompt combo.

**Comment by DrakenZA** [comment]
![myoutpainting](https://user-images.githubusercontent.com/8331354/189529588-29d1b798-6da6-441c-af80-064252a3fd5a.PNG)

My attempt at outpainting via GIMP and my personal plugin that talks to this projects code.
